File: data/src/main/java/io/forus/me/android/data/entity/vouchers/response/Transaction.java
package io.forus.me.android.data.entity.vouchers.response;

import com.google.gson.annotations.SerializedName;

import java.math.BigDecimal;
import java.util.Date;

public class Transaction {

    @SerializedName("id")
    private Long id;

    @SerializedName("organization_id")
    private Long organizationId;

    @SerializedName("product_id")
    private Long productId;

    @SerializedName("amount")
    private BigDecimal amount;

    @SerializedName("amount_extra_cash")
    private BigDecimal amount_extra_cash;

    @SerializedName("address")
    private String address;

    @SerializedName("created_at")
    private Date createdAt;

    @SerializedName("updated_at")
    private Date updatedAt;

    @SerializedName("date")
    private String date;

    @SerializedName("timestamp")
    private Long timestamp;

    @SerializedName("organization")
    private Organization organization;

    @SerializedName("product")
    private Product product;

    @SerializedName("fund")
    private Fund fund;


    @SerializedName("state")
    private String state;


    public Transaction() { }

    public Transaction(Long id, Long organizationId, Long productId, BigDecimal amount, BigDecimal amount_extra_cash,
                       String address, Date createdAt, Date updatedAt, String date, Long timestamp,
                       Organization organization, Product product, Fund fund, String state) {
        this.id = id;
        this.organizationId = organizationId;
        this.productId = productId;
        this.amount = amount;
        this.amount_extra_cash = amount_extra_cash;
        this.address = address;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
        this.date = date;
        this.timestamp = timestamp;
        this.organization = organization;
        this.product = product;
        this.fund = fund;
        this.state = state;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getOrganizationId() {
        return organizationId;
    }

    public void setOrganizationId(Long organizationId) {
        this.organizationId = organizationId;
    }

    public Long getProductId() {
        return productId;
    }

    public void setProductId(Long productId) {
        this.productId = productId;
    }

    public BigDecimal getAmount() {
        return amount;
    }

    public void setAmount(BigDecimal amount) {
        this.amount = amount;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    public Date getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Date updatedAt) {
        this.updatedAt = updatedAt;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public Long getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(Long timestamp) {
        this.timestamp = timestamp;
    }

    public Organization getOrganization() {
        return organization;
    }

    public void setOrganization(Organization organization) {
        this.organization = organization;
    }

    public Product getProduct() {
        return product;
    }

    public void setProduct(Product product) {
        this.product = product;
    }

    public Fund getFund() {
        return fund;
    }

    public void setFund(Fund fund) {
        this.fund = fund;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public BigDecimal getAmount_extra_cash() {
        return amount_extra_cash;
    }
}

File: domain/src/main/java/io/forus/me/android/domain/models/vouchers/Transaction.kt
package io.forus.me.android.domain.models.vouchers

import java.math.BigDecimal
import java.util.*

class Transaction {

    var id: String

    var organization: Organization?

    var amount: BigDecimal?

    var amount_locale: String?

    var amount_extra_cash: BigDecimal?

    var amount_extra_cash_locale: String?

    var createdAt: Date?

    var type: Type = Type.Payed

    var product: Product?

    var state: String?

    var fund: Fund?

    var note: String?

    constructor(
        id: String,
        organization: Organization?,
        amount: BigDecimal?,
        amount_locale: String?,
        amount_extra_cash: BigDecimal?,
        amount_extra_cash_locale: String?,
        createdAt: Date?,
        product: Product?,
        state: String?,
        fund: Fund?,
        note: String?
    ) {
        this.id = id
        this.organization = organization
        this.amount = amount
        this.amount_locale = amount_locale
        this.amount_extra_cash = amount_extra_cash
        this.amount_extra_cash_locale = amount_extra_cash_locale
        this.createdAt = createdAt
        this.type = type
        this.product = product
        this.state = state
        this.fund = fund
        this.note = note
    }


    enum class Type {
        Payed, Refund, Cancel, Income, Product
    }
}
